source/PlayerInfo.cpp

```diff
@@ -2774,6 +2799,35 @@ void PlayerInfo::AddStock(const Outfit *outfit, int count)
 
 
 
+// Transfer ships from the player to the planet or vice versa.
+// If needed, create a no-outfit version of this ship if we don't already have one.
+void PlayerInfo::AddStock(const Ship *model, const int count)
+{
+	shipStock[model] += count;
+
+	if(shipStock[model] <= 0 || stockShips.find(model) != stockShips.end())
```

Comment:
count can be negative (happens when the player buys stuff), so shipStock[model] can also go negative.  We only have to worry about setting up the ship info if the player's net selling stuff.

Reply:
what about the second check?
I find it ironic you check it after referencing it too x)

Reply:
Ah, those are two different variables with very similar names (shipStock and stockShips).  One tracks number of ships in stock and the other one is a list of stocked ship objects.

Reply:
~~that's pretty confusing~~
why do we need both? and could you find more distinct names mb?

Reply:
stockShips are Ship objects with no outfits (as opposed to the GameData ships, which have default outfits).  It's relevant because outfits modify various ship attributes, so when DrawDetails is called, having outfits or not changes everything that gets displayed, not just the list of outfits.

So, while the player is at a planet, we keep a list of ship types that the player has net sold to the store, so we can display them properly in the Shipyard.

However, we also need to know how many of those ships got sold, which is what gets tracked by shipStock.  When that hits zero for a given ship model, then we go back to using the default shop models (or just don't display it, if the store doesn't sell it).

Both names make sense on their own, but I won't deny that they can be confusing, particularly when next to each other.

Reply:
I'd rename the second one to soldShips? it's not only stock ones if I understand well

Reply:
I went with shipChassis, since that's really what they are.